### PR TITLE
Ripley spider

### PR DESCRIPTION
**Pull request by Rodp63**

### CODE REVIEW

File: ripley/ripley/spiders/ripley_product_spider.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        brand = response.xpath('//*[contains(@class,"brand-logo")]/span/text()').extract_first()        
+        iPrice = response.xpath('//*[contains(@itemprop,"lowPrice")]/text()').extract_first()
+        nPrice = response.xpath('//*[contains(@itemprop,"price")]/text()').extract()[0]
+        
+        yield{
+            'Code':  response.xpath('//*[contains(@itemprop,"sku")]/text()').extract_first(),
+            'Brand': brand if brand else "Not found",
+            'Description': response.xpath('//*[contains(@class,"product-short-description")]/text()').extract_first(), #exception
+            'Internet_Price': iPrice if iPrice else "Not found",
+            'Normal_Price': nPrice if nPrice else "Not found",
```
[the remark this answers is not in the document]
Comment (joaquingx) [reply ripley/ripley/spiders/ripley_product_spider.py]: Maybe you could try loaders for items?https://github.com/joaquingx/PEShopSpiders/blob/master/ShopSpiders/ShopSpiders/loaders/shop_loaders.py

File: ripley/ripley/spiders/ripley_product_spider.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    name = "ripley"
+    sitemap_urls = ['https://home.ripley.com.pe/products_sitemap.xml']
+    allowed_domains = ['simple.ripley.com.pe']
+    
+    def parse(self, response):
+        
+        brand = response.xpath('//*[contains(@class,"brand-logo")]/span/text()').extract_first()        
+        iPrice = response.xpath('//*[contains(@itemprop,"lowPrice")]/text()').extract_first()
+        nPrice = response.xpath('//*[contains(@itemprop,"price")]/text()').extract()[0]
```
[the remark this answers is not in the document]
Comment (Rodp63) [reply ripley/ripley/spiders/ripley_product_spider.py]: Just to try :D

File: ripley/ripley/spiders/ripley_categories_spider.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            iPrice = item.xpath('.//li[@class="catalog-prices__offer-price"]/text()').extract_first()
+            discount = item.xpath('.//div[@class="catalog-product-details__discount-tag"]/text()').extract_first()
+            yield{
+                'Brand': brand if brand else "Not found",
+                'Product_name': name,
+                'Normal_price': nPrice if nPrice else "Not found",
+                'Internet_price': iPrice if iPrice else "Not found",
+                'Discount': discount if discount else "None",
+            }
+        mark = response.xpath('//*[@id="catalog-page"]/div/div[2]/div[4]/nav/ul/li[4]/a/@href').extract_first()
```
[the remark this answers is not in the document]
Comment (Rodp63) [reply ripley/ripley/spiders/ripley_categories_spider.py]: Fixed

File: ripley/ripley/spiders/ripley_product_spider.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+class RipleySpider(SitemapSpider):
+
+    name = "ripley"
+    sitemap_urls = ['https://home.ripley.com.pe/products_sitemap.xml']
+    allowed_domains = ['simple.ripley.com.pe']
+    
+    def parse(self, response):
+        
+        brand = response.xpath('//*[contains(@class,"brand-logo")]/span/text()').extract_first()
```
[the remark this answers is not in the document]
Comment (Rodp63) [reply ripley/ripley/spiders/ripley_product_spider.py]: I did not notice that. Thanks, brand name is always there!

**Comment by joaquingx** [comment]
Great Job @Rodp63, thanks :pray:!

**Pull request merged**